feat(theme): final re-brand

kiaking opened the pull request:
- close #2610 

The final re-branding before going RC. The VitePress brand is now aligns with Vite.

I have tweak mostly the CSS vars but tiny other parts that I wonted adjust a bit. Now we have 2 separate colors for Light/Dark mode. We should be eliminate a11y problems we have or would have in the future.

- [x] Add social preview images

---

<img width="1392" alt="Screenshot 2023-08-03 at 21 13 51" src="https://github.com/vuejs/vitepress/assets/3753672/8c49e951-b98b-4270-82ff-f5d5441a0357">

<img width="1392" alt="Screenshot 2023-08-03 at 21 14 05" src="https://github.com/vuejs/vitepress/assets/3753672/1b4e0adf-c814-41d6-b515-94c5d6bf8126">

<img width="612" alt="Screenshot 2023-08-03 at 21 16 00" src="https://github.com/vuejs/vitepress/assets/3753672/1d1aa96c-9ce0-4d21-b9e3-93560ab9efe5">

<img width="612" alt="Screenshot 2023-08-03 at 21 15 40" src="https://github.com/vuejs/vitepress/assets/3753672/a669bf30-2603-442e-b128-bd924dad4c86">

brc-dd reviewed (approved):
updated some stuff to improve perf 🚀

kiaking commented on src/client/theme-default/styles/vars.css:
```diff
@@ -86,31 +58,67 @@
-
   --vp-c-mute: #f6f6f7;
   --vp-c-mute-light: #f9f9fc;
   --vp-c-mute-lighter: #ffffff;
```
Updated! 👍

kiaking commented:
Thanks for the tweak! I don't know why but I converted png to webp and it got even smaller (33kb) so I've replaced it with that 🤔  And fixed some border color issue on CSS var.

zonemeen commented:
Great great great job! Look so cool !👀

brc-dd commented:
> I don't know why but I converted png to webp and it got even smaller (33kb)

Yeah, I too expected it to be smaller. But pixelmator generated that 50 kB or something.

Lehoczky commented:
Shall we adjust the text color for the danger containers `(--vp-custom-block-danger-text)` to meet the AA contrast ratio? 

I found the following suggestions using [chrome devtools](https://developer.chrome.com/docs/devtools/accessibility/contrast/#fix-low-contrast), that are very close to the current color:
- for **dark** theme: `#ff4666` instead of `#f43f5e` 
- for **light** theme: `#cd354f` instead of `#f43f5e`

Right now we use the same color for both light and dark themes, but unfortunately there is no shade of red which meets the AA contrast requirement for both themes:

![image](https://github.com/vuejs/vitepress/assets/31937175/04616f26-a059-4d19-8397-51ede4caa158)
![image](https://github.com/vuejs/vitepress/assets/31937175/3c9932ce-f3fa-464a-9dc3-6c8f7d1b381d)

First picture is for dark, second is for light theme.

brc-dd commented:
@kiaking close #2119 as well if the container colors are adjusted 🙌

PS: we can also pick colors from https://www.radix-ui.com/colors, their grouping seem to yield 5-10 contrast ratio (use 9/10/11 for text, 1/2 for bg)

Lehoczky commented on src/client/theme-default/styles/components/vp-doc.css:
```diff
@@ -109,13 +109,10 @@
 .vp-doc a:hover {
-  /* color: var(--vp-c-brand-dark); */
-  text-decoration: underline;
+  text-decoration: underline dotted;
```
Just a suggestion, I think moving the dotted line down a bit looks a bit better. What's do your opinion?

```suggestion
  text-decoration: underline dotted;
  text-underline-offset: 2px;
```

![image](https://github.com/vuejs/vitepress/assets/31937175/0897a23b-5f52-490d-8ce9-a1c55b013056)
![image](https://github.com/vuejs/vitepress/assets/31937175/88d7a3ab-9e6e-4b0b-a0e5-4aadaceaf491)

userquin commented:
@brc_dd do we have some codeblock inside `::: <tip|info|warning|error>` in the playground? We should check color constrast, something like this https://vite-pwa-org.netlify.app/guide/auto-update.html#automatic-reload.

brc-dd commented:
> do we have some codeblock inside `::: <tip|info|warning|error>` in the playground?

By playground, do you mean e2e tests? There aren't a11y checks there right now. We can add those, but currently they will fail. They shouldn't be necessary IMO, we won't later be changing the colors much.

userquin commented:
I mean, with a simple visual test va dev `tools.,` there is no need to add `a11y` tests

brc-dd commented:
What's playground? The deploy preview https://deploy-preview-2727--vitepress-docs.netlify.app/guide/markdown#custom-containers ?

kiaking commented:
Added social preview image 👍
